components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/MonetizationApiServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+        try {
+            APIAdmin apiAdmin = new APIAdminImpl();
+            monetizationUsagePublishInfo = apiAdmin.getMonetizationUsagePublishInfo();
+            if (monetizationUsagePublishInfo == null) {
+                monetizationUsagePublishInfo = new MonetizationUsagePublishInfo();
+                monetizationUsagePublishInfo.setId(APIConstants.MonetizationUsagePublisher.JOB_NAME);
+                monetizationUsagePublishInfo.setState(APIConstants.MonetizationUsagePublisher.INITIATED);
+                monetizationUsagePublishInfo.setStatus(APIConstants.MonetizationUsagePublisher.INPROGRESS);
+                //read the number of days to reduce from the current time to derive the from / last publish time
+                //when there is no record of the last publish time
+                APIManagerConfiguration configuration = ServiceReferenceHolder.getInstance().
+                        getAPIManagerConfigurationService().getAPIManagerConfiguration();
+                String gap = configuration.getFirstProperty(
+                        APIConstants.MonetizationUsagePublisher.FROM_TIME_CONFIGURATION_PROPERTY);
+                //if the from time / last publish time is not set , set it to default
+                if (gap == null) {
+                    gap = APIConstants.MonetizationUsagePublisher.DEFAULT_TIME_GAP_IN_DAYS;
+                }
+                DateFormat df = new SimpleDateFormat(APIConstants.MonetizationUsagePublisher.TIME_FORMAT);
+                df.setTimeZone(TimeZone.getTimeZone(APIConstants.MonetizationUsagePublisher.TIME_ZONE));
+                Calendar cal = Calendar.getInstance();
+                Date currentDate = cal.getTime();
+                String formattedCurrentDate = df.format(currentDate);
+                long currentTimestamp = apiAdmin.getTimestamp(formattedCurrentDate);
+                monetizationUsagePublishInfo.setStartedTime(currentTimestamp);
+                //reducing the number of days set to get the last published time when there is no record of
+                //the last published time
+                cal.add(Calendar.DATE, -Integer.parseInt(gap));
+                Date fromDate = cal.getTime();
+                String formattedFromDate = df.format(fromDate);
+                long lastPublishedTimeStamp = apiAdmin.getTimestamp(formattedFromDate);
+                monetizationUsagePublishInfo.setLastPublishTime(lastPublishedTimeStamp);
+                apiAdmin.addMonetizationUsagePublishInfo(monetizationUsagePublishInfo);
+            }
+            if (!monetizationUsagePublishInfo.getState().equals(APIConstants.MonetizationUsagePublisher.RUNNING)) {
+                executor = Executors.newSingleThreadExecutor();
+                MonetizationUsagePublishAgent agent = new MonetizationUsagePublishAgent(monetizationUsagePublishInfo);
+                executor.execute(agent);
+                String staus = "Request Accepted";
```

Comment:
aren't the variable status and msg redundant? since it's only used once and contains 2 string values

Reply:
The API DTO returns two values, status and the message of the publishing job. This is available for any custom implementation to return the above values separately in the API response.

Reply:
no what i meant was like 
`return Response.accepted().entity(MonetizationAPIMappinUtil.fromStatusToDTO("Request Accepted", "Server is running the usage publisher")).build();
`

Reply:
Fixed in 0cb422c3d49f3c8e3e3d4e02e156665fd004836b